### PR TITLE
Use ID_LIKE for most preflights checks

### PR DESCRIPTION
**Pull request by guillaumerose**
This should fix some derivative distros of Ubuntu (popos, linux mint).

### CODE REVIEW

File: pkg/crc/preflight/preflight_linux.go [hunk 1]
```diff
@@ -218,14 +218,6 @@ func usesSystemdResolved(distro *linux.OsRelease) bool {
 	}
 }
 
-func distroID(osRelease *linux.OsRelease) linux.OsType {
-	if osRelease == nil {
```
Comment (cfergeau) [inline pkg/crc/preflight/preflight_linux.go:222]: `distro()` can return `nil` if `linux.GetOsRelease()` fails, so it's possible to get a `nil` `*linux.OsRelease` here.

File: pkg/crc/preflight/preflight_checks_linux.go [hunk 2]
```diff
@@ -144,10 +144,10 @@ func fixLibvirtInstalled(distro *linux.OsRelease) func() error {
 
 func installLibvirtCommand(distro *linux.OsRelease) string {
 	yumCommand := "yum install -y libvirt libvirt-daemon-kvm qemu-kvm"
-	switch distroID(distro) {
-	case linux.Ubuntu:
+	switch {
+	case distroIsLike(distro, linux.Ubuntu):
 		return "apt-get update && apt-get install -y libvirt-daemon libvirt-daemon-system libvirt-clients"
-	case linux.RHEL, linux.CentOS, linux.Fedora:
+	case distroIsLike(distro, linux.Fedora), distroIsLike(distro, linux.RHEL):
```
Comment (cfergeau) [inline pkg/crc/preflight/preflight_checks_linux.go:150]: fwiw, RHEL and CentOS have `ID_LIKE=fedora`

File: pkg/crc/preflight/preflight_linux.go (hunk 1 above)
Comment (guillaumerose) [reply pkg/crc/preflight/preflight_linux.go:222]: outch, thanks!

Comment (guillaumerose) [reply pkg/crc/preflight/preflight_linux.go:222]: done

File: pkg/crc/preflight/preflight_checks_linux.go [hunk 3]
```diff
@@ -144,10 +144,10 @@ func fixLibvirtInstalled(distro *linux.OsRelease) func() error {
...
 func installLibvirtCommand(distro *linux.OsRelease) string {
 	yumCommand := "yum install -y libvirt libvirt-daemon-kvm qemu-kvm"
-	switch distroID(distro) {
-	case linux.Ubuntu:
+	switch {
+	case distroIsLike(distro, linux.Ubuntu):
 		return "apt-get update && apt-get install -y libvirt-daemon libvirt-daemon-system libvirt-clients"
-	case linux.RHEL, linux.CentOS, linux.Fedora:
+	case distroIsLike(distro, linux.Fedora):
 		return yumCommand
```
Comment (cfergeau) [inline pkg/crc/preflight/preflight_checks_linux.go:151]: While you are changing this method, you could probably remove the `yumCommand` variable which doesn't bring much, and is inconsistent with the ubuntu case.

**Review by cfergeau — APPROVED** [review]
I would move the 'Always return an OsRelease struct, even if /etc/os-release is absent ' commit before `Assume systemd resolved is used by all derivative distros of Ubuntu` and `Remove distroID(), use directly distro.ID` as they both are dependent on this change.

**Review by cfergeau — APPROVED** [review]

File: pkg/crc/preflight/preflight_checks_linux.go (hunk 3 above)
Comment (guillaumerose) [reply pkg/crc/preflight/preflight_checks_linux.go:151]: I changed the order but for yumCommand, I would have to duplicate in the fedora check and in the default check. Perhaps you have an other idea?

**Pull request merged**